src/once_cell.rs

```diff
@@ -71,21 +71,39 @@ impl<T> GILOnceCell<T> {
             return value;
         }
 
+        match self.init(py, || Ok::<T, std::convert::Infallible>(f())) {
+            Ok(value) => value,
+            Err(void) => match void {},
+        }
+    }
+
+    /// Like `get_or_init`, but accepts a fallible initialization function.
+    ///
+    /// If it fails, the cell is left uninitialized.
```

Comment:
Can you expand on what reentrant initialization does here? (Does it deadlock like the `once_cell` impl?) And what about panics?

Reply:
We should probably centralize the documentation already present on `get_or_init` at the type level to avoid repeating ourselves here.

Reply:
Agreed, I've done this now.